src/hooks/useColors.ts

```diff
@@ -0,0 +1,105 @@
+import { useThemeConfigurationContext } from '../providers/ThemeConfigurationProvider';
+
+/* eslint-disable */
+const Gradient = require('javascript-color-gradient');
+/* eslint-enable */
+
+export const usePrimary = (gradientValue: number) => {
+  const { color } = useThemeConfigurationContext();
+  if (gradientValue < 1) return color.PRIMARY;
+  const gradient = new Gradient()
```

Comment:
@RiceCooker-dev It may be necessary to mutualized code because a lot of things are currently repeated